### ISSUE TITLE
Is there a way to redirect gurobi's log into julia logging 

### ISSUE THREAD

**Issue by Betristor** [jump-dev/Gurobi.jl#487]
Typically julia's output (mostly println and minor logging messages) will show in the REPL, but is there a way to redirect gurobi's log into logging too, as ```Logging``` and ```LoggingExtras``` modules will do.

**Comment by odow** [comment]
I'm not sure if there is an easy way to redirect to `Logging.jl`, but I _guess_ you could write a solver-specific callback that intercepted all `MESSAGE` callbacks:
```Julia
using JuMP, Gurobi
model = Model(Gurobi.Optimizer)
set_optimizer_attribute(model, "LogToConsole", false)
function my_callback_function(cb_data, cb_where::Cint)
    if cb_where == GRB_CB_MESSAGE
        resultP = Ref{Ptr{Cchar}}()
        GRBcbget(cb_data, cb_where, GRB_CB_MSG_STRING, resultP) 
        @info unsafe_string(resultP[])
    end
    return
end
MOI.set(model, Gurobi.CallbackFunction(), my_callback_function)

julia> optimize!(model)
[ Info: Gurobi Optimizer version 9.5.1 build v9.5.1rc2 (mac64[x86])
[ Info: Thread count: 4 physical cores, 8 logical processors, using up to 8 threads
[ Info: Optimize a model with 0 rows, 0 columns and 0 nonzeros
[ Info: Model fingerprint: 0xf9715da1
[ Info: Coefficient statistics:
[ Info:   Matrix range     [0e+00, 0e+00]
[ Info:   Objective range  [0e+00, 0e+00]
[ Info:   Bounds range     [0e+00, 0e+00]
[ Info:   RHS range        [0e+00, 0e+00]
[ Info: Presolve time: 0.00s
[ Info: Presolve: All rows and columns removed
[ Info: Iteration    Objective       Primal Inf.    Dual Inf.      Time
[ Info:        0    0.0000000e+00   0.000000e+00   0.000000e+00      0s
[ Info: 
[ Info: Solved in 0 iterations and 0.00 seconds (0.00 work units)
[ Info: Optimal objective  0.000000000e+00
[ Info: 
[ Info: User-callback calls 20, time in user-callback 0.03 sec
```

See also:

 * https://www.gurobi.com/documentation/9.5/refman/logfile.html#parameter:LogFile
 * https://www.gurobi.com/documentation/9.5/refman/logtoconsole.html#parameter:LogToConsole

**Comment by Betristor** [comment]
Thanks a lot. I implemented this function by passing ```LogFile``` parameter into gurobi's setting when using JuMP.
